pkgs/development/tools/parsing/antlr/4.nix

```
@@ -0,0 +1,157 @@
+{ lib, stdenv, fetchurl, jre
+, fetchFromGitHub, cmake, ninja, pkg-config, darwin
+
+# ANTLR 4.8 & 4.9
+, libuuid
+
+# ANTLR 4.9
+, utf8cpp }:
+
+let
+
+  mkAntlr = { version, sourceSha256, jarSha256 }: lib.fixedPoints.makeExtensible (self: {
```

```suggestion
  mkAntlr = { version, sourceSha256, jarSha256 }: {
```
Do we really need this? Usually this is enough.